### ISSUE TITLE
Gradient infill

### ISSUE THREAD

**Issue by hetile-ssense** [Ultimaker/Cura#6954]
As demonstrated by CNC Kitchen in this video, gradient infill on the XY would be a very nice (and probably easy) feature to add to Cura.
https://www.youtube.com/watch?v=hq53gsYREHU

He provided open source python script here:
https://github.com/CNCKitchen/GradientInfill

This would make cura the only slicer to support it built-in without having to hack anything.

**Comment by LarsSimonsen** [comment]
![image](https://user-images.githubusercontent.com/1157543/72218589-a32a7500-353c-11ea-844b-7e841143972e.png)
The concept is brilliantly simple. 

Btw, it has already been implemented in [PrusaSlicer as an experimental build](https://www.reddit.com/r/prusa3d/comments/enjx4d/i_added_support_for_cnc_kitchens_gradient_infill/).

**Comment by hetile-ssense** [comment]
I think it would be best to have it implemented as a full infill pattern otherwise it might messup Linear Advance (marlin) or Pressure Advance (klipper)

**Comment by mahtDFR** [comment]
This also caught our attention this weekend. His script adjusts extrusion rather than speed, so is optimized for direct drive machines. Bowden machines running Marlin will operate a little differently and need more tuning. We will continue to investigate...

**Comment by Firedrops** [comment]
That method might not work on more complicated infills like gyroid, which probably requires the whole geometry to be re-generated with smoothly varied densities.

**Comment by Cugio** [comment]
Yes, this could be very useful, and of course will probably work only with selected infill type.

**Comment by 5axes** [comment]
For testing purpose I've converted CNC Kitchen Python script into a Cura Postprocessing plug in Script.
![PlugIn](https://user-images.githubusercontent.com/11015345/72688053-110a0a00-3b04-11ea-8725-f602c0e98951.jpg)

Unfortunalty as mentionned by @mahtDFR  ( and also in the CNC Kitchen publication )  This solution seams to be not very efficient on a machine Bowden machines running Marlin. If some one wants to make more teste on a different machine : [https://github.com/5axes/GradientInfill/blob/master/GradientInfill.py](url)

**Comment by hetile-ssense** [comment]
The suggestion would be more of having this infill pattern as another type of infill without playing with the extrusion.

**Comment by 5axes** [comment]
> Bowden machines running Marlin will operate a little differently and need more tuning. We will continue to investigate...

Using the gradient Infill on a machine whitout a direct drive must definetly manage a speed variation to get a correct result. 
![82570108_1223017127904648_3642722292435255296_o](https://user-images.githubusercontent.com/11015345/72863579-65aeb000-3cd1-11ea-8fb9-26b74ead47e3.jpg)

Sample test print with a Gradiant distance of 8 mm and a infill percentage of 45%
![82574446_1223039984569029_7656888964539744256_o](https://user-images.githubusercontent.com/11015345/72863512-284a2280-3cd1-11ea-8d7c-5500abe6417a.jpg)

**Comment by Liger0** [comment]
It would also be different between Marlin's Linear Advance users and not I guess, that has to be considered as there is a Cura LA extension. 
Also LA can be used on both bowden and Direct drive.

**Comment by zbyrek** [comment]
I wonder, why you consider changing speed\flow rather than making paths closer to eachother at edges (like setting infill 100%) and make them more further apart at center (like with infill 20% or so) when generating infil, that should work without problems in every setting (bowden\direct\linear advance).

**Comment by MattLogue** [comment]
6 Vectors would be nice as cura only does top, cnc kitchen does xy+/-. Sadly, the ender3 does NOT support relative extrusion which is required for some archaic reason. I could redo it for absolute units I suppose...
